src/dynamicprompts/samplers/utils.py

```diff
@@ -50,3 +63,55 @@ def get_wildcard_not_found_fallback(
     res = SamplingResult(text=wrapped_wildcard)
     while True:
         yield res
+
+
+def replace_wildcard_variables(
+    command: WildcardCommand,
+    *,
+    context: SamplingContext,
+) -> WildcardCommand:
+    if not command.wildcard.__contains__(context.parser_config.variable_start):
+        return command
+
+    prompt = pp.SkipTo(context.parser_config.variable_end)
+    variable_access = _configure_variable_access(
+        parser_config=context.parser_config,
+        prompt=prompt,
+    )
+    variable_access.set_parse_action(
+        partial(_replace_variable, variables=context.variables),
+    )
+    wildcard = _configure_wildcard_path(
+        parser_config=context.parser_config,
+        variable_ref=variable_access,
+    )
+
+    try:
+        wildcard_result = wildcard.parse_string(command.wildcard)
+        return command.with_content("".join(wildcard_result))
+    except Exception as ex:
+        logger.warning("Unable to parse wildcard path %s", command.wildcard)
+        return command
+
+
+def _replace_variable(string, location, token, *, variables: dict):
+    if isinstance(token, pp.ParseResults):
+        var_parts = token[0].as_dict()
+        var_name = var_parts.get("name")
+        if var_name:
+            var_name = var_name.strip()
+
+        default = var_parts.get("default")
+        if default:
+            default = default.strip()
+
+    else:
+        var_name = token
+
+    variable = None
+    if var_name:
+        variable = variables.get(var_name)
+
+        if isinstance(variable, LiteralCommand):
+            variable = variable.literal
```

Comment:
What happens if the variable is _not_ a `LiteralCommand` here? (You can assign other commands than just literals to variables, and they are generally then evaluated when they are substituted.)

We should probably evaluate them here, or maybe, for the time being, add a

```python
if not isinstance(variable, str):
    raise NotImplementedError("evaluating complex commands within wildcards is not supported right now")
```
and come back to that in a subsequent PR :)

Reply:
I added the check requested. I am curious as to what other types of expressions can appear within variables.

Reply:
E.g.

https://github.com/adieyal/dynamicprompts/blob/510d1e958c26276512188532c6cfae4306d291d7/tests/test_variable_features.py#L10

shows a variant command being assigned into a variable, and since it's not `!` immediate evaluated (see notes in https://github.com/adieyal/dynamicprompts/pull/63) it will remain that way until actually substituted into a prompt.